core/src/main/java/org/mvcspec/ozark/core/ViewResponseFilter.java

```diff
@@ -173,6 +179,24 @@ public void filter(ContainerRequestContext requestContext,
         }
     }
 
+    /*
+     * Append to view name default extension if one available and applicable.
+     */
+    protected static String appendExtensionIfRequired(String viewName, String defaultExtension) {
+        if (viewName == null || viewName.startsWith(REDIRECT)
+            || defaultExtension == null || defaultExtension.length() == 0) {
+            return viewName;
+        }
+
+        String resultView = viewName;
+        if (!viewName.contains(DOT_CHAR) && !viewName.contains(":")) {
+            resultView += DOT_CHAR + defaultExtension;
+        } else if (viewName.endsWith(DOT_CHAR)) {
+            throw new IllegalArgumentException("Invalid format for view name: " + viewName);
```

Comment:
I don't think that we should fail here. Of course view names with trailing dots are weird, but technically they will work. So why not removing this else block?

Reply:
Else block removed.